cns/restserver/errors.go

```diff
@@ -0,0 +1,18 @@
+package restserver
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/Azure/azure-container-networking/cns/types"
+)
+
+var errResponseCode = errors.New("Response code is error")
+
+// ResponseCodeToError converts a cns response code to error type. If the response code is OK, then return value is nil
+func ResponseCodeToError(responseCode types.ResponseCode) error {
```

Comment:
I would suggest creating a new type that implmenets `error`, such as `CNSRESTError`, which might look like:
```
type CNSRESTError struct {
    ResponseCode types.ResponseCode
}
func (c *CNSRESTError) Error() string {
    return fmt.Errorf("response code: %d", c.ResponseCode)
}
```
This allows errors to hold data better, rather than just storing all the data in the error string. If we start using the `errors.Is` and `errors.As` functions, this will be a big help.

Reply:
Of course, then this `ResponseCodeToError` function becomes
```
return &CNSRESTError{
    ResponseCode: responseCode,
}
```

Reply:
Thanks @thatmattlong , wouldn't we still want an if statement in `ResponseCodeToError` like so?
```
func ResponseCodeToError(responseCode types.ResponseCode) error {
	if responseCode == types.Success {
		return nil
	}

	return &CNSRESTError{
		ResponseCode: responseCode,
	}
}

```

Reply:
_really_ we would want to go the other way, and return the error struct or nil where we are currently returning the response code, but that might snowball this change considerably

Reply:
Yeah I took a peek and it did seem to unravel a bit

Reply:
I will resolve this unless anyone has more comments on it

Reply:
ack